clang/runtime/dpct-rt/include/dpct/dpl_extras/algorithm.h

```diff
@@ -978,7 +1018,41 @@ stable_partition(Policy &&policy, Iter1 first, Iter1 last, Iter2 mask, Pred p) {
 }
 
 template <typename Policy, typename Iter1, typename Iter2, typename Pred>
-internal::enable_if_execution_policy<Policy, Iter1>
+typename std::enable_if<internal::is_hetero_execution_policy<
+                            typename std::decay<Policy>::type>::value,
+                        Iter1>::type
+partition(Policy &&policy, Iter1 first, Iter1 last, Iter2 mask, Pred p) {
+  static_assert(
+      std::is_same<typename std::iterator_traits<Iter1>::iterator_category,
+                   std::random_access_iterator_tag>::value &&
+          std::is_same<typename std::iterator_traits<Iter2>::iterator_category,
+                       std::random_access_iterator_tag>::value,
+      "Iterators passed to algorithms must be random-access iterators.");
+  typedef typename std::decay<Policy>::type policy_type;
```

Comment:
I don't think this line is used anywhere.

Reply:
fixed